### PR TITLE
Iot agent

### PR DESCRIPTION
**Pull request by taliaga**

### CODE REVIEW

File: recipes/iot-services/iotagent-json/config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    host: 'mosquitto',
+    port: 1883,
+    thinkingThingsPlugin: true
+};
+
+config.iota = {
+    logLevel: 'DEBUG',
+    timestamp: true,
+    contextBroker: {
+        host: 'orion',
```
Comment (chicco785) [inline recipes/iot-services/iotagent-json/config.js]: anyway to make this an environment variable? it is supposed to be deployed in a different recipe.

File: recipes/iot-services/iotagent-json/config.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    },
+    server: {
+        port: 4041
+    },
+    deviceRegistry: {
+        type: 'mongodb'
+    },
+    mongodb: {
+        host: 'mongo',
+        port: '27017',
```
Comment (chicco785) [inline recipes/iot-services/iotagent-json/config.js]: anyway to make this an environment variable? it is supposed to be deployed in a different recipe.

File: recipes/iot-services/iotagent-json/config.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        db: 'iotagentjson'
+    },
```
Comment (chicco785) [inline recipes/iot-services/iotagent-json/config.js]: also, how to support a replicaset?

File: recipes/iot-services/iotagent-json/test/add_sleep.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+curl -X POST -H "Content-Type: application/json" -H "Accept: application/json" -H "Fiware-Service: myHome" -H "Fiware-ServicePath: /environment" -H "Cache-Control: no-cache" -d '{"type" : "configuration", "value" : "300"}' "http://${ORION_EP}/v1/contextEntities/LivingRoomSensor/attributes/sleepTime"
```
Comment (chicco785) [inline recipes/iot-services/iotagent-json/test/add_sleep.sh]: missing CR at the end of the command

File: recipes/iot-services/iotagent-json/test/setup.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+export ORION_EP=192.168.99.100:1026
+export IOTA_EP=192.168.99.100:4041
```
Comment (chicco785) [inline recipes/iot-services/iotagent-json/test/setup.sh]: CR missing

File: recipes/iot-services/iotagent-lwm2m/config.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            name: 'Robot',
+            url: '/robots'
+        }
+    ]
+};
+
+config.ngsi = {
+    logLevel: 'DEBUG',
+    contextBroker: {
+        host: 'orion',
```
Comment (chicco785) [inline recipes/iot-services/iotagent-lwm2m/config.js]: should be env variable

File: recipes/iot-services/iotagent-lwm2m/config.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        port: '1026'
+    },
```
Comment (chicco785) [inline recipes/iot-services/iotagent-lwm2m/config.js]: should be env variable

File: recipes/iot-services/iotagent-lwm2m/config.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    server: {
+        port: 4041
+    },
+    deviceRegistry: {
+        type: 'memory'
+    },
```
Comment (chicco785) [inline recipes/iot-services/iotagent-lwm2m/config.js]: why not mongo?

File: recipes/iot-services/iotagent-lwm2m/test/device_provisioning/provision_device.sh [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+            "objectType": 7392,
+            "objectInstance": 0,
+            "objectResource": 3
+          }
+        }
+      }
+    }
+  ]
+}
+EOF
```
Comment (chicco785) [inline recipes/iot-services/iotagent-lwm2m/test/device_provisioning/provision_device.sh]: CR missing

File: recipes/iot-services/iotagent-lwm2m/test/device_provisioning/query_broker.sh [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+{
+    "entities": [
+        {
+            "type": "Robot",
+            "isPattern": "false",
+            "id": "Robot:robot1"
+        }
+    ]
+}
+EOF
```
Comment (chicco785) [inline recipes/iot-services/iotagent-lwm2m/test/device_provisioning/query_broker.sh]: CR missing

File: recipes/iot-services/iotagent-lwm2m/test/device_provisioning/send_command.sh [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+                "name": "Position",
+                "type": "location",
+                "value": "[18,3]"
+            }
+            ]
+        }
+    ],
+    "updateAction": "UPDATE"
+}
+EOF
```
Comment (chicco785) [inline recipes/iot-services/iotagent-lwm2m/test/device_provisioning/send_command.sh]: CR missing

File: recipes/iot-services/iotagent-lwm2m/test/provisioning_config/provision_config.sh [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+      "active": [
+        {
+          "name": "Power Control",
+          "type": "Boolean"
+        }
+      ]
+    }
+  ]
+}
+EOF
```
Comment (chicco785) [inline recipes/iot-services/iotagent-lwm2m/test/provisioning_config/provision_config.sh:35]: CR missing

File: recipes/iot-services/iotagent-lwm2m/test/provisioning_config/query_lazy.sh [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+            "type": "WeatherBaloon",
+            "isPattern": "false",
+            "id": "weather1:WeatherBaloon"
+        }
+    ],
+    "attributes" : [
+        "Power Control"
+    ]    
+}
+EOF
```
Comment (chicco785) [inline recipes/iot-services/iotagent-lwm2m/test/provisioning_config/query_lazy.sh]: CR missing

File: recipes/iot-services/iotagent-ul/config.js [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+config.http = {
+    port: 7896
+};
+
+config.iota = {
+    logLevel: 'DEBUG',
+    timestamp: true,
+    contextBroker: {
+        host: 'orion',
+        port: '1026'
```
Comment (chicco785) [inline recipes/iot-services/iotagent-ul/config.js]: orion and its port should be an env variable

File: recipes/iot-services/iotagent-ul/config.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+        port: 4041
+    },
+    deviceRegistry: {
+        type: 'mongodb'
+    },
+    mongodb: {
+        host: 'mongo',
+        port: '27017',
+        db: 'iotagentul'
+    },
```
Comment (chicco785) [inline recipes/iot-services/iotagent-ul/config.js]: 1. mongo and port should be an env variable
2. support for replicaset

File: recipes/iot-services/iotagent-ul/test/register_sensor.sh [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+         "name":      "luminosity",
+         "type":      "number"
+       }
+     ]
+   }
+ ]
+}'
+
+# Check
+curl -H "Content-type: application/json" -H "Fiware-Service: openiot" -H "Fiware-ServicePath: /" http://${IOTA_EP}/iot/devices | jq .
```
Comment (chicco785) [inline recipes/iot-services/iotagent-ul/test/register_sensor.sh]: CR missing

File: recipes/iot-services/iotagent-ul/test/register_service.sh [hunk 17]
```diff
@@ -0,0 +1,19 @@
+#/bin/bash
+
+# Register
+curl -H "Content-type: application/json" -H "Fiware-Service: openiot" -H "Fiware-ServicePath: /" http://${IOTA_EP}/iot/services -d '{
+ "services": [
+   {
+     "apikey":      "4jggokgpepnvsb2uv4s40d59ov",
+     "cbroker":     "http://orion:1026",
```
Comment (chicco785) [inline recipes/iot-services/iotagent-ul/test/register_service.sh:8]: env variable?

**Review by chicco785 — CHANGES_REQUESTED** [review]
in general:
1. introduce ENV variables
2. support for replica-set seems not included

File: recipes/iot-services/iotagent-lwm2m/config.js (hunk 8 above)
Comment (taliaga) [reply recipes/iot-services/iotagent-lwm2m/config.js]: It was just the default in the original repo. Updated to use mongo by default.

**Comment by taliaga** [comment]
@chicco785 updated with env variables config (including replicaset name) and empty lines fixes

**Review by chicco785 — APPROVED** [review]

**Comment by chicco785** [comment]
LGTM, I wonder if there is any way to disable replicaset if need, empty value?

**Comment by taliaga** [comment]
Empty value would do, removing the variable also. Added comment on this. Thanks!

**Pull request merged**